### PR TITLE
Ruevla 15 & Ruelvla 17

### PR DESCRIPTION
**Pull request by beahuszar**
both stories completed in this pull request

### CODE REVIEW

File: rueppellii-lvlup-asp.net-core/Controllers/BadgesController.cs [hunk 1]
```diff
@@ -11,27 +11,21 @@ namespace rueppellii_lvlup_asp.net_core.Controllers
...
         }
-        return Unauthorized(new ErrorMessage("Unauthorized"));
+
+        [HttpGet("badges")]
+        public IActionResult ListBadges()
+        {
+            if (Request.Headers["usertokenauth"] == "gen")
+            {
+                return Ok(service.GetAll());
+            }
```
Comment (drexeverest) [inline rueppellii-lvlup-asp.net-core/Controllers/BadgesController.cs]: The else is not mandatory for this logic to be executed

**Review by drexeverest — APPROVED** [review]

File: rueppellii-lvlup-asp.net-core/Services/Interfaces/IReadService.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+namespace rueppellii_lvlup_asp.net_core.Services
+{
+    /// <summary>
+    /// U = dto
+    /// T = entity
+    /// Can be extended with get methods by different properties
+    /// </summary>
+    public interface IReadService<U, T> where U : class where T : class
+    {
+        IEnumerable<U> GetAll(); //gets all the entities from the repo, returns a JSON list
```
Comment (vadam85) [inline rueppellii-lvlup-asp.net-core/Services/Interfaces/IReadService.cs]: This does not return a JSON list, but an iterable collection of DTO-s. (Or am i misunderstanding something?)

Comment (beahuszar) [reply rueppellii-lvlup-asp.net-core/Services/Interfaces/IReadService.cs]: resolved :)

File: rueppellii-lvlup-asp.net-core/Controllers/BadgesController.cs (hunk 1 above)
Comment (beahuszar) [reply rueppellii-lvlup-asp.net-core/Controllers/BadgesController.cs]: 👍

**Review by drexeverest — APPROVED** [review]

**Pull request merged**